tests/gpflow/conditionals/test_multioutput.py

```diff
@@ -728,3 +729,48 @@ def test_separate_independent_conditional_with_q_sqrt_none():
         q_sqrt=q_sqrt,
         white=True,
     )
+
+
+def test_independent_interdomain_conditional_bug_regression():
+    """
+    Regression test for https://github.com/GPflow/GPflow/issues/818
+    Not an exhaustive test
+    """
+    M = 32
+    N = 10
+    D = 36 * 2
+
+    Zs = [np.random.randn(M, D // 2) for _ in range(2)]
+    k = gpflow.kernels.SquaredExponential(D // 2)
+
+    def compute_Kmn(Z, X):
+        X1 = X[:, 0 : D // 2]
+        X2 = X[:, D // 2 :]
+        return tf.stack([k(Z, X1), k(Z, X2)])
+
+    def compute_Knn(X):
+        X1 = X[:, 0 : D // 2]
+        X2 = X[:, D // 2 :]
+        return tf.stack([k(X1, full_cov=False), k(X2, full_cov=False)])
+
+    Xbatch = np.random.randn(N, D)
+    # L = 2, P = 2, N = 10, M = 32
```

Comment:
can we set L different to P? If so, would probably be a slightly better test to check this case.

Reply:
the independent model requires the number of latent GPs L to be equivalent to the number of outputs P, no?

Reply:
I've fixed the test accordingly! (We could still do with better tests - see #1295)